swing/src/net/sf/openrocket/gui/rocketfigure/RailButtonShapes.java

```diff
@@ -40,45 +49,78 @@
 		final double flangeHeightcos = flangeHeight*cosr;
 
 		Path2D.Double path = new Path2D.Double();
+		Path2D.Double pathInvis = new Path2D.Double();	// Path for the invisible triangles
 		{// central pillar
 			final double drawWidth = outerDiameter;
 			final double drawHeight = outerDiameter*sinr;
-			final Point2D.Double center = new Point2D.Double( instanceAbsoluteLocation.x, instanceAbsoluteLocation.y );
+			final Point2D.Double center = new Point2D.Double( loc.x, loc.y );
 			Point2D.Double lowerLeft = new Point2D.Double( center.x - outerRadius, center.y-outerRadius*sinr);
 			path.append( new Ellipse2D.Double( lowerLeft.x, lowerLeft.y, drawWidth, drawHeight), false);
 			
 			path.append( new Line2D.Double( lowerLeft.x,  center.y, lowerLeft.x, (center.y+baseHeightcos) ), false);
 			path.append( new Line2D.Double( (center.x+outerRadius),  center.y, (center.x+outerRadius), (center.y+baseHeightcos) ), false);
 			
 			path.append( new Ellipse2D.Double( lowerLeft.x, (lowerLeft.y+baseHeightcos), drawWidth, drawHeight), false);
+
+			// Invisible rectangle
+			double y_invis;
+			if (baseHeightcos >= 0)
```

Comment:
another nit: curly braces always, even when optional... why? The following is valid, but it is difficult to chase down those missing {}

```java
if (5 == bar)
    do_something();
    do_something_else();
```